### PR TITLE
Fix ref warning on React 19

### PR DESCRIPTION
**Pull request by joshwooding**
Bumps React to 18.3.3
Adds getRefFromChildren to read refs from children in a way that supports all version of react. Also reduces the amount of ts-ignores/expect errors.
Updated Accordion, Menu, Overlay, Tooltip, CascadingMenu and Portal.

### CODE REVIEW

File: packages/core/src/utils/getRefFromChildren.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { isValidElement, ReactNode } from "react";
+
+export function getRefFromChildren(child: ReactNode) {
```
Comment (DavieReid) [inline packages/core/src/utils/getRefFromChildren.ts:3]: Does this need a test?

File: packages/lab/src/cascading-menu/CascadingMenu.tsx [hunk 2]
```diff
@@ -195,9 +194,7 @@ export const CascadingMenu = forwardRef<HTMLDivElement, CascadingMenuProps>(
       childrenRef.current = node;
     }, []);
     const handleRef = useForkRef(
-      //TODO get to bottom of this
-      // @ts-ignore
-      isValidElement(children) ? children.ref : noop,
```
Comment (DavieReid) [inline packages/lab/src/cascading-menu/CascadingMenu.tsx:200]: This one returned `noop` not `null` like in the other usages.  

The update looks Ok but just pointing out the diff

**Review by DavieReid — APPROVED** [review]

File: packages/lab/src/cascading-menu/CascadingMenu.tsx (hunk 2 above)
Comment (joshwooding) [reply packages/lab/src/cascading-menu/CascadingMenu.tsx:200]: Yeah, i think this was just an old way of doing it

File: packages/core/src/utils/getRefFromChildren.ts (hunk 1 above)
Comment (joshwooding) [reply packages/core/src/utils/getRefFromChildren.ts:3]: Added :)

**Review by mark-tate — APPROVED** [review]

**Review by DavieReid — APPROVED** [review]

**Pull request merged**